### ISSUE TITLE
Print commit hash with `bun --revision`

### ISSUE THREAD

**Issue by lgarron** [oven-sh/bun#3676]
### What is the problem this feature would solve?

While testing https://github.com/oven-sh/bun/issues/3669 I've been using `bun upgrade --canary`, which is much more convenient than building `bun` from scratch (especially if you haven't pulled `main` since a bunch of dependencies changed).

However, I don't see an obvious way to find the version of a currently installed canary.

### What is the feature you are proposing to solve the problem?

I think it would be really handy if `bun --version` printed enough information to identify an exact project commit.

> git describe --tags
bun-v0.6.14-53-g8bd2b784a

One option might be to modify https://github.com/oven-sh/bun/blob/3ec6c9e4fb03a374bb5a1d856bf804483246589c/.github/workflows/bun-release.yml#L68C1-L71 to use:

```shell
> git describe --tags
bun-v0.6.14-53-g8bd2b784a
```

Since this runs after `actions/checkout@v3`, I think it shouldn't be more expensive than what it's currently doing.

I'm on macOS Ventura 13.4.1, if it helps to know.

**Comment by lgarron** [comment]
> Since this runs after `actions/checkout@v3`, I think it shouldn't be more expensive than what it's currently doing.

Hmm, it looks like the relevant code might actually be trying to do something like this:

https://github.com/oven-sh/bun/blob/d265ed80d20267bc5252c31022af36d139bc283e/.github/workflows/bun-release-types-canary.yml#L38-L42

Nevertheless, it doesn't show up for me when I run `bun --version` (and I'm definitely seeing behaviour right now that is only fixed in the latest canary).

**Comment by Electroid** [comment]
We can't change `bun --version`, because old versions of Bun depend on its format for upgrades. But we can introduce something new like `bun --revision` that includes the git commit.

```
> bun --version
0.7.2
> bun --revision
0.7.2+g8bd2b78
```

**Comment by hyuzea** [comment]
currently on linux i can use 
```
echo 'console.info(Bun.revision)' | bunx bun-repl 
[0.07ms] ".env"
Welcome to Bun.js v0.7.2
Type ".help" for more information.
928816bef0945e9655c115767c719d6865ff138c
```
or 
```
echo 'console.info(Bun)' | bunx bun-repl | grep revision
[0.04ms] ".env"
  revision: "928816bef0945e9655c115767c719d6865ff138c",
```

**Comment by lgarron** [comment]
@yus-ham Thanks, that's super handy!

I think this issue is still relevant for a discoverable command, but that's certainly useful. 😄 

(Although it would be nice to avoid relying on a package, but I can't seem to get `bun run (echo "console.info(Bun.revision)" | psub)` to work in `fish`.)

**Comment by birkskyum** [comment]
this seems to be closed by #4027

**Comment by lgarron** [comment]
> this seems to be closed by #4027

Looks like it!

I think matching `git describe --tags` would have been nice, but it's certainly nice to have an easy way to get the hash!
